lib/localisation/src/filtering/kalman_filter.rs

```diff
@@ -0,0 +1,144 @@
+use nalgebra::{DMatrix, DVector};
+
+pub struct KalmanFilter {
+    // Current state estimate
+    state: DVector<f64>,
+
+    // Current error covariance
+    covariance: DMatrix<f64>,
+
+    // State transition matrix
+    transition_matrix: DMatrix<f64>,
+
+    // Control matrix
+    control_matrix: DMatrix<f64>,
+
+    // Observation matrix
+    observation_matrix: DMatrix<f64>,
+
+    // Process noise covariance
+    process_noise: DMatrix<f64>,
+
+    // Measurement noise covariance
+    measurement_noise: DMatrix<f64>,
+}
+
+impl KalmanFilter {
+    pub fn new(
+        initial_state: DVector<f64>,
+        initial_covariance: DMatrix<f64>,
+        transition_matrix: DMatrix<f64>,
+        control_matrix: DMatrix<f64>,
+        observation_matrix: DMatrix<f64>,
+        process_noise: DMatrix<f64>,
+        measurement_noise: DMatrix<f64>,
+    ) -> Self {
+        KalmanFilter {
+            state: initial_state,
+            covariance: initial_covariance,
+            transition_matrix,
+            observation_matrix,
+            control_matrix,
+            process_noise,
+            measurement_noise,
+        }
+    }
+
+    // Predict step
```

Comment:
```suggestion
    /// Predict step
```
Might be good also to quickly explain what it's doing to predict (what is the prediction based on?)